templates/common/baremetal/files/baremetal-keepalived.yaml

```diff
@@ -57,29 +34,79 @@ contents:
           - name: NSS_SDB_USE_CACHE
             value: "no"
         command:
-        - /usr/sbin/keepalived
-        args:
-        - "-f"
-        - "/etc/keepalived/keepalived.conf"
-        - "--dont-fork"
-        - "--vrrp"
-        - "--log-detail"
-        - "--log-console"
+        - /bin/bash
+        - -c
+        - |
+          #/bin/bash
+          reload_keepalived()
+          {
+            declare -r pid="$(pgrep -o keepalived)"
+            kill -s SIGHUP "$pid"
+          }
+
+          msg_handler()
+          {
+            while read -r line; do
+              echo "The client sent: $line"  >&2
+              # currently only 'reload' msg is supported
+              if [ "$line" = reload ]; then
+                  reload_keepalived
+              fi
+            done
+          }
+
+          set -ex
+          declare -r keepalived_sock="/var/run/keepalived/keepalived.sock"
+          export -f msg_handler
+          export -f reload_keepalived
+          if [ -s "/etc/keepalived/keepalived.conf" ]; then
+              /usr/sbin/keepalived -f /etc/keepalived/keepalived.conf --dont-fork --vrrp --log-detail --log-console &
+          fi
+
+          rm -f "$keepalived_sock"
```

Comment:
You remove it and start listening to it on the next line? Why?

Reply:
In case the container is restarted the socket file might be there and socat won't be able to start unless it is gone.

Reply:
Ack